source/agora/flash/Node.d

```diff
@@ -884,20 +884,29 @@ public abstract class FlashNode : ControlFlashAPI
 
             auto chans = (*path).map!(hop => hop.chan_id);
             OnionError deobfuscated = err;
-            foreach (secret; shared_secrets)
+            size_t failing_hop_idx = size_t.max;
+            foreach (idx, secret; shared_secrets)
             {
                 if (chans.canFind(deobfuscated.chan_id))
+                {
+                    failing_hop_idx = idx;
                     break;
+                }
                 deobfuscated = deobfuscated.obfuscate(secret);
             }
-
-            if (chans.canFind(deobfuscated.chan_id))
-            {
-                log.info(this.conf.key_pair.address.flashPrettify, " Got error: ",
-                    deobfuscated);
-                this.payment_errors[deobfuscated.payment_hash] ~= deobfuscated;
-                this.dump();
-            }
+            if (failing_hop_idx == size_t.max)
+                return;
+
+            const failing_node_pk = (*path)[failing_hop_idx].pub_key;
+            const failing_chan = this.known_channels[deobfuscated.chan_id];
+            if (failing_chan.funder_pk != failing_node_pk &&
```

Comment:
Some `//` comments here about what the code is doing would be great.

Reply:
Done, made a small change to the loop above too/